Client/Frontend/Browser/Playlist/Controllers/PlaylistFolderController.swift

```diff
@@ -0,0 +1,595 @@
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        // Reload the table only when completely visible
+        tableView.reloadData()
+    }
+    
+    @objc
+    private func onDonePressed(_ button: UIBarButtonItem) {
+        self.dismiss(animated: true)
+    }
+}
+
+extension PlaylistFolderController: UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return Section.allCases.count
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        guard let section = Section(rawValue: section) else {
+            return 0
+        }
+        
+        switch section {
+        case .savedItems: return 1
+        case .folders: return othersFRC.fetchedObjects?.count ?? 0
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier) else {
+            return UITableViewCell(style: .subtitle, reuseIdentifier: cellIdentifier)
+        }
+        
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        
+        guard let section = Section(rawValue: indexPath.section) else {
+            return
+        }
+        
+        let folderIcon = UIImage(systemName: "folder")?.template
+        
+        switch section {
+        case .savedItems:
+            let itemCount = savedFolder?.playlistItems?.count ?? 0
+            
+            cell.imageView?.image = folderIcon
+            cell.textLabel?.text = Strings.PlaylistFolders.playlistSavedFolderTitle
+            cell.detailTextLabel?.text = "\(itemCount) \(itemCount == 1 ? Strings.PlaylistFolders.playlistFolderSubtitleItemSingleCount : Strings.PlaylistFolders.playlistFolderSubtitleItemCount)"
+            cell.detailTextLabel?.textColor = .secondaryBraveLabel
+            cell.accessoryType = .disclosureIndicator
+            cell.selectionStyle = .none
+        case .folders:
+            guard let folder = othersFRC.fetchedObjects?[safe: indexPath.row] else {
+                return
+            }
+            
+            let itemCount = folder.playlistItems?.count ?? 0
+            
+            cell.imageView?.image = folderIcon
+            cell.textLabel?.text = folder.title
+            cell.detailTextLabel?.text = "\(itemCount) \(itemCount == 1 ? Strings.PlaylistFolders.playlistFolderSubtitleItemSingleCount : Strings.PlaylistFolders.playlistFolderSubtitleItemCount)"
+            cell.detailTextLabel?.textColor = .secondaryBraveLabel
+            cell.accessoryType = .disclosureIndicator
+            cell.selectionStyle = .none
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        52
```

Comment:
Why do we have static cell height? It will not be possible to use dynamic height here ?

Reply:
I'm keeping it the exact same as I got it from the figma designs.
We don't need dynamic cell heights because we want all the cells the same height.

Reply:
All cells will stay the same height but dynamic fonts will make it overall taller: 
![image](https://user-images.githubusercontent.com/529104/155233229-abcd81a4-8ab5-4a0c-9e33-f5654aedaec5.png)